### ISSUE TITLE
Takes 8.2053734e+57 years or longer to complete - can we speed it up? tl;dr: no silly, you can't

### ISSUE THREAD

**Issue by CurtisASmith** [VectorsCrib/OpenaiAPIKeyMiner#1]
Sorry to ruin your hopes, but math wants to say hi. An atheist who knew the odds would start praising God if this worked for them. Put simply, someone winning the Powerball lottery twice in a row is improbable to occur in our lifetime, yet possible; finding a working key for this API with this technique would be improbable to occur over the lifetimes of a decillion universes, utterly impossible in a practical sense. The numbers don't lie, check it out:

```
61^40 = 2.5893658e+71 # This is the number of possibilities.
2.5893658e+71/1e6 = 2.5893658e+65 # How many seconds it would take to check each possibility once at 1 million per second
2.5893658e+65/31556952 = 8.2053734e+57 # Divide by seconds in a year for total number of years
8.2053734e+57/1.38e+10 = 5.9459228e+47 divide by age of the universe in years and realize how big those numbers are.
```

Maybe you have a massive botnet and you believe somehow you can check faster than a million times per second. I'm sure you know you can't, but just in case, math's buddy physics wants you to remember that there are physical limits to network connections, and you'll only manage to DDoS OpenAI if you tried. Not to mention how difficult it is to ensure you don't waste time guessing the same permutation twice, which the current code definitely doesn't check for. Trying to write a list for that would be screwed by space limitations. There's another way to implement your technique to get around that, but it wouldn't solve any of the other problems, and would introduce other ones you aren't ready for.

I understand the desire to access GPT-3 for yourself, but this isn't the way to do it. Give up, sign up for the waitlist like the rest of us, or donate resources to help build open recreations like [Eleuther's GPT-NeoX](https://www.eleuther.ai/).

If this did work for you already, let me know, I'll be your cult's first disciple 😂

**Comment by VectorsCrib** [comment]
Yes, this is just a stupid script that has no value to be on github. I will be removing it shortly unless I decide I want to rework it a bit.
